wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java

```diff
@@ -0,0 +1,275 @@
+          DevfileDto devfile,
+      @HeaderParam(CONTENT_TYPE) MediaType contentType)
+      throws ConflictException, BadRequestException, ForbiddenException, NotFoundException,
+          ServerException {
+    requiredNotNull(devfile, "Devfile");
+    return Response.status(201)
+        .entity(
+            linksInjector.injectLinks(
+                asDto(userDevfileManager.createDevfile(devfile)), getServiceContext()))
+        .build();
+  }
+
+  @GET
+  @Path("/{id}")
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(value = "Get devfile by its identifier")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The response contains requested workspace entity"),
+    @ApiResponse(code = 404, message = "The workspace with specified id does not exist"),
+    @ApiResponse(code = 403, message = "The user is not allowed to read devfile"),
+    @ApiResponse(code = 500, message = "Internal server error occurred")
+  })
+  public UserDevfileDto getById(
+      @ApiParam(value = "UserDevfile identifier") @PathParam("id") String id)
+      throws NotFoundException, ServerException, ForbiddenException, BadRequestException {
+    requiredNotNull(id, "id");
+    return linksInjector.injectLinks(asDto(userDevfileManager.getById(id)), getServiceContext());
+  }
+
+  @GET
+  @Path("list")
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(
+      value = "Get devfiles which user can read",
+      notes =
+          "This operation can be performed only by authorized user. "
+              + "It is possible to add additional constraints for the desired devfiles by specifying\n"
+              + "multiple query parameters that is representing fields of the devfile. All constrains\n"
+              + "would be combined with \"And\" condition. Also, it is possible to specify 'like:' prefix\n"
+              + "for the query parameters. In this case instead of an exact match would be used SQL pattern like search.\n"
+              + "Examples id=sdfsdf5&devfile.meta.name=like:%dfdf&",
+      response = UserDevfileDto.class,
+      responseContainer = "List")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The devfiles successfully fetched"),
+    @ApiResponse(code = 500, message = "Internal server error occurred during devfiles fetching")
+  })
+  public Response getUserDevfiles(
+      @ApiParam("The number of the items to skip") @DefaultValue("0") @QueryParam("skipCount")
+          Integer skipCount,
+      @ApiParam("The limit of the items in the response, default is 30, maximum 60")
+          @DefaultValue("30")
+          @QueryParam("maxItems")
+          Integer maxItems,
+      @ApiParam(
+              "A list of fields and directions of sort. By default items would be sorted by id.Example id:asc,devfile.metadata.name desc.")
+          @QueryParam("order")
+          String order)
+      throws ServerException, BadRequestException {
+    if (maxItems > 60) {
```

Comment:
Where does this limit come from? For example `/workspace` endpoint doesn't seem to impose it.